### DISCUSSION TITLE
Find Productdata

### DISCUSSION THREAD

**Discussion by ChrisSantiago82** [lunarphp/lunar#1389] [Q&A]
I'm new the lunar and I have the following question:

I'm trying to import our products and products variants from our POS system. This should not be a one time thing, this should happen automatically. If there is a new product, it should automatically appear in Lunar.

My issue is, how can I check if the product already is created, and I just need to add the product variant, or if I have to create a new product?

In the product table, there is only the "attribute_data" field, and it doesn't seem a good way to make queries against a json field. Like from the POS, I have an unique key that I would like to use for the search.
Or is there going to be any issue if I just add a field to this table?

![image](https://github.com/lunarphp/lunar/assets/43000085/2a1a4536-5fa2-4198-93fd-5a32e9f6968d)

**Comment by glennjacobs** [comment]
I would imagine you would check for the SKU on the product variant.

**Comment by ChrisSantiago82** [comment]
Yes, but lets say I have 10 variants. And I have and id from the POS that binds them together. That would be the product in Lunar.
Maybe a week later I get 5 more for the same product. Now I need to check if that Product already exists.

And thank you so much for helping me :)

**Reply by glennjacobs** [reply]
Right, so yes your options are...

1) Use custom attributes with JSON where clause
2) Add a new model/table to store the mapping
3) Add a new field to the products table and extend the product model

**Reply by ChrisSantiago82** [reply]
Yes, that is what I was thinking.

Thank you
